### PR TITLE
Documentation reorg

### PR DESCRIPTION
**Pull request by mccalluc**
I have been merging smaller feature branches into this, and just merged `main` into it to bring it up to date.

~~Waiting to move out of draft until after the 0.9 release.~~ Talked with Raman: Given the uncertain timeline on polars, and the trouble we'll have trying to merge doc changes into this, going to document the system as it currently exists, and try to merge this sooner than later.

--- 

Update:
- Last child branches reviewed and merged
- up to date
- Built docs locally and skimmed every page
- Fixed some typos, and filed a half dozen new, hopefully small issues that shouldn't block this.

This is a big update, and not without risk: Most of the urls on our docs site are changing, and links from outside are likely to break. -- but, I think this is a lot cleaner, and however people get here, they are more likely to find what they need. Most of the changes here were their own PRs that already were reviewed.

### CODE REVIEW

File: docs/source/api/user-guide/combinators/index.rst [hunk 1]
```diff
@@ -27,37 +27,45 @@ In this case, it is only necessary for the types to conform.
...
-    ...
-    >>> # investigate the privacy relation
-    >>> symmetric_distance = 1
-    >>> epsilon = 1.0
-    >>> assert noisy_sum.check(d_in=symmetric_distance, d_out=epsilon)
-    ...
-    >>> # invoke the chained measurement's function
-    >>> dataset = [0, 0, 1, 1, 0, 1, 1, 1]
-    >>> release = noisy_sum(dataset)
+.. tabs::
```
Comment (Shoeboxam) [inline docs/source/api/user-guide/combinators/index.rst:30]: For me, these tabs don't render well on the dark theme.

File: docs/source/getting-started/hello-opendp.rst [hunk 2]
```diff
@@ -1,5 +1,5 @@
-Quickstart
-==========
+Hello OpenDP!
```
Comment (Shoeboxam) [inline docs/source/getting-started/hello-opendp.rst:1]: I think "Quickstart" is still better.

File: docs/source/getting-started/hello-opendp.rst [hunk 3]
```diff
@@ -38,17 +38,13 @@ Enable ``contrib`` globally with the following snippet:
```
Comment (Shoeboxam) [inline docs/source/getting-started/hello-opendp.rst:38]: The `install.packages` link above should go to `repos = 'https://opendp.r-universe.dev/'`

**Review by Shoeboxam — APPROVED** [review]
* I would think _Limitations_ and _Installation_ would go under _Getting Started_ and _Programming Framework_ would go under _Theory_.
* Given that we've moving towards tabs, can you rename Python API User Guide to just User Guide?
* Rename _Getting Started_ under _Theory_, as we already have a top-level section named _Getting Started_.
* Can _Real World_ be renamed to something like _Release Workflow_, and be adjusted to use tabs, where the tabs correspond to the Python Framework API and Python Context API? 

Thanks for all the work on this!

**Review by Shoeboxam — DISMISSED** [review]
Request changes. :P

File: docs/source/api/user-guide/combinators/index.rst (hunk 1 above)
Comment (mccalluc) [reply docs/source/api/user-guide/combinators/index.rst:30]: I hadn't checked that -- Good catch.

File: docs/source/getting-started/hello-opendp.rst (hunk 2 above)
Comment (mccalluc) [reply docs/source/getting-started/hello-opendp.rst:1]: Will revert.

**Comment by mccalluc** [comment]
> * I would think _Limitations_ and _Installation_ would go under _Getting Started_ 

I really want to keep _Getting Started_ tight, and get Python and R users started as quickly as possible. There's some redundancy here. A proposal:
- Add tabs for Rust on _Getting Started_ .
- and after that get rid of _Installation_? The only other section it has is "Building OpenDP from Source", but that's really only of interest to devs, and they can find the developer instructions.

I think `Limitations` is important -- not trying to hide it! -- But I thought it fit better at the start of the _API User Guide_, when people are starting to get into the details, rather than at the end of _Getting Started_, which is more focused on demos.

> ... and _Programming Framework_ would go under _Theory_.

For me, the _Theory_ section is about the mathematics of DP in general, sometimes with examples of OpenDP, but not just the theory of OpenDP.

API_ centers on OpenDP itself, and thinking that way, _Programming Framework_ with its details about OpenDP feels like it belongs here.

> * Given that we've moving towards tabs, can you rename Python API User Guide to just User Guide?

Sure!

> * Rename _Getting Started_ under _Theory_, as we already have a top-level section named _Getting Started_.

Maybe rename to _Implementing DP with OpenDP_? or _Doing DP with OpenDP_? For context, the previous page is _A Framework to Understand DP_

> * Can _Real World_ be renamed to something like _Release Workflow_, and be adjusted to use tabs, where the tabs correspond to the Python Framework API and Python Context API? 

Maybe just _Workflow_? Particularly since this is the first section, if I didn't already know DP, and I saw "Release" in the TOC, I would be thinking about software releases.

**Comment by mccalluc** [comment]
Changes made -- Not everything you suggested, but if my reasoning above isn't persuasive, I'm probably just not quite understanding where you're coming from: Let's find time to talk.

Marking as draft while there are TODOs:

TODOs:
- [ ] Rust example: laplace on a scalar
- [ ] CI to confirm that Rust example works
- [ ] Clarify goals for tabbed context/framework examples

Right now we have:

> Context API:
> > ```
> > >>> import opendp.prelude as dp
> > 
> > >>> # we are also using library features that are marked "contrib":
> > >>> dp.enable_features("contrib")
> > 
> > >>> privacy_unit = dp.unit_of(contributions=1)
> > ```
> 
> Framework API:
> > The privacy unit is actually a 2-tuple:
> > ```
> > >>> import opendp.prelude as dp
> > 
> > >>>> # we are also using library features that are marked "contrib":
> > >>> dp.enable_features("contrib")
> > 
> > >>> input_metric, d_in = privacy_unit
> > 
> > >>> assert d_in == 1 # neighboring data set distance is at most d_in...
> > >>> assert input_metric == dp.symmetric_distance() # ...in terms of additions/removals
> > ```
> > The privacy unit tuple specifies how distances are computed between two data sets (``input_metric``), and how large the distance can be (``d_in``).

This made sense when the Framework discussion followed the Context discussion... but now that they are in parallel, we want to demo Framework without using any variables described in the other track. Would this make sense?

> Framework API:
> > ```
> > >>> import opendp.prelude as dp
> > 
> > >>>> # we are also using library features that are marked "contrib":
> > >>> dp.enable_features("contrib")
> > 
> > >>> d_in = 1 # neighboring data set distance is at most d_in...
> > >>> input_metric = dp.symmetric_distance() # ...in terms of additions/removals

There's no longer a destructuring assignment, but maybe it doesn't belong here any longer: For the Context API, you don't need to take apart the tuple, and with just the framework, you don't have the utility to give you the tuple.

**Comment by mccalluc** [comment]
Last run has an R test failure -- sporadic?

**Comment by mccalluc** [comment]
Section 5 doesn't have all the description that it formerly had. If we want to add more, I'd suggest doing it as comments inside the code, instead of trying having more inclusion blocks we need to keep track of.

Even if this passes review, should wait until the `base_` refactor is in before merging.

**Review by raprasad — APPROVED** [review]
Approving the updates.

File: docs/source/getting-started/hello-opendp.rst (hunk 2 above)
Comment (raprasad) [reply docs/source/getting-started/hello-opendp.rst]: - moved to #1346

File: docs/source/getting-started/hello-opendp.rst (hunk 3 above)
Comment (raprasad) [reply docs/source/getting-started/hello-opendp.rst]: resolved

**Pull request merged**